File: ios/TourGraph/TourGraph/TourGraph/Models/Superlative.swift
import Foundation

enum SuperlativeType: String, CaseIterable, Sendable {
    case mostExpensive = "most-expensive"
    case cheapest5Star = "cheapest-5star"
    case longest
    case shortest
    case mostReviewed = "most-reviewed"
    case hiddenGem = "hidden-gem"

    var displayTitle: String {
        switch self {
        case .mostExpensive: "Most Expensive"
        case .cheapest5Star: "Cheapest 5-Star"
        case .longest: "Longest"
        case .shortest: "Shortest"
        case .mostReviewed: "Most Reviewed"
        case .hiddenGem: "Hidden Gem"
        }
    }

    var emoji: String {
        switch self {
        case .mostExpensive: "💎"
        case .cheapest5Star: "🏷️"
        case .longest: "⏳"
        case .shortest: "⚡"
        case .mostReviewed: "🗣️"
        case .hiddenGem: "🌟"
        }
    }

    var statLabel: String {
        switch self {
        case .mostExpensive: "price"
        case .cheapest5Star: "price"
        case .longest: "duration"
        case .shortest: "duration"
        case .mostReviewed: "reviews"
        case .hiddenGem: "rating"
        }
    }
}

struct SuperlativeResult: Identifiable, Sendable {
    let type: SuperlativeType
    let tour: Tour
    var id: String { type.rawValue }
}

File: android/TourGraph/app/src/main/java/com/nikhilsi/tourgraph/model/Superlative.kt
package com.nikhilsi.tourgraph.model

enum class SuperlativeType(
    val displayTitle: String,
    val emoji: String,
    val statLabel: String
) {
    MOST_EXPENSIVE("Most Expensive", "\uD83D\uDCB0", "Price"),
    CHEAPEST_5_STAR("Cheapest 5-Star", "\u2B50", "Price"),
    LONGEST("Longest", "\u23F3", "Duration"),
    SHORTEST("Shortest", "\u26A1", "Duration"),
    MOST_REVIEWED("Most Reviewed", "\uD83D\uDDE3\uFE0F", "Reviews"),
    HIDDEN_GEM("Hidden Gem", "\uD83C\uDF1F", "Rating");
}

data class SuperlativeResult(
    val type: SuperlativeType,
    val tour: Tour
)
